deps/rabbit/src/rabbit_peer_discovery.erl

```diff
@@ -108,29 +108,22 @@ maybe_init() ->
     %% node, even if the configuration changed in between.
     persistent_term:put(?PT_PEER_DISC_BACKEND, Backend),
 
-    _ = code:ensure_loaded(Backend),
-    case erlang:function_exported(Backend, init, 0) of
-        true  ->
-            ?LOG_DEBUG(
-               "Peer discovery: backend supports initialisation",
+    case catch Backend:init() of
```

Comment:
The use of a `catch` here means that other exceptions will become a tuple, causing a `case_clause` exception below. Therefore we lose the real root exception and in particular its location.

I would prefer a regular `try ... catch`. What do you think?